src/CashuWallet.ts

```diff
@@ -47,87 +41,232 @@ import {
+ */
+const DEFAULT_DENOMINATION_TARGET = 3;
+
+/**
+ * The default unit for the wallet, if not specified in constructor.
+ */
+const DEFAULT_UNIT = 'sat';
+
 /**
  * Class that represents a Cashu wallet.
  * This class should act as the entry point for this library
  */
 class CashuWallet {
-	private _keys: MintKeys | undefined;
-	private _seed: Uint8Array | undefined;
-	private _unit = 'sat';
+	private _keys: Map<string, MintKeys> = new Map();
+	private _keysetId: string | undefined;
+	private _keysets: Array<MintKeyset> = [];
+	private _seed: Uint8Array | undefined = undefined;
+	private _unit = DEFAULT_UNIT;
+	private _mintInfo: GetInfoResponse | undefined = undefined;
+	private _denominationTarget = DEFAULT_DENOMINATION_TARGET;
+
 	mint: CashuMint;
 
 	/**
-	 * @param unit optionally set unit
-	 * @param keys public keys from the mint. If set, it will override the unit with the keysets unit
 	 * @param mint Cashu mint instance is used to make api calls
-	 * @param mnemonicOrSeed mnemonic phrase or Seed to initial derivation key for this wallets deterministic secrets. When the mnemonic is provided, the seed will be derived from it.
+	 * @param options.unit optionally set unit (default is 'sat')
+	 * @param options.keys public keys from the mint (will be fetched from mint if not provided)
+	 * @param options.keysets keysets from the mint (will be fetched from mint if not provided)
+	 * @param options.mintInfo mint info from the mint (will be fetched from mint if not provided)
+	 * @param options.denominationTarget target number proofs per denomination (default: see @constant DEFAULT_DENOMINATION_TARGET)
+	 * @param options.mnemonicOrSeed mnemonic phrase or Seed to initial derivation key for this wallet's deterministic secrets. When the mnemonic is provided, the seed will be derived from it.
 	 * This can lead to poor performance, in which case the seed should be directly provided
+	 * @param options.loadMint if set to true info will be loaded from mint
 	 */
 	constructor(
 		mint: CashuMint,
 		options?: {
 			unit?: string;
-			keys?: MintKeys;
+			keys?: Array<MintKeys> | MintKeys;
+			keysets?: Array<MintKeyset>;
+			mintInfo?: GetInfoResponse;
 			mnemonicOrSeed?: string | Uint8Array;
+			denominationTarget?: number;
+			loadMint?: boolean;
 		}
 	) {
 		this.mint = mint;
+		let keys: Array<MintKeys> = [];
+		if (options?.keys && !Array.isArray(options.keys)) {
+			keys = [options.keys];
+		} else if (options?.keys && Array.isArray(options?.keys)) {
+			keys = options?.keys;
+		}
+		if (keys) keys.forEach((key: MintKeys) => this._keys.set(key.id, key));
 		if (options?.unit) this._unit = options?.unit;
-		if (options?.keys) {
-			this._keys = options.keys;
-			this._unit = options.keys.unit;
+		if (options?.keysets) this._keysets = options.keysets;
+		if (options?.denominationTarget) {
+			this._denominationTarget = options.denominationTarget;
+		}
+
+		if (options?.loadMint) {
+			this.loadMint()
```

Comment:
we should either add events here, or do it do it async outside of the constructor. right now, we don't know when the mint has been loaded

Reply:
I agree. Methods that require a up to date key set could call this method and await its return. Potentially with a cache invalidation logic

Reply:
> Methods that require a up to date key set could call this method and await its return. Potentially with a cache invalidation logic

out of scope

Reply:
> Methods that require a up to date key set could call this method and await its return.

They do: `await loadKeys()` when needed etc.

> Potentially with a cache invalidation logic

out of scope

Reply:
Then this should be removed from the constructor.

Reply:
Done